### PR TITLE
Added frontend Input Field Validations on the EventForm.tsx and Enabled validations in app.ts

### LINKED ISSUE #430: Validate URL Fields like events.event_url and events.video_url
Per #423 and #424, we agreed to add validation to check the user provided input is a valid URL.

It was suggested by @timmyichen [`new URL()` wasn't the best solution](https://github.com/freeCodeCamp/chapter/pull/423#discussion_r512192170) and we [agreed](https://github.com/freeCodeCamp/chapter/pull/423#discussion_r512199357) to merge in the database fields, but [create a new issue](https://github.com/freeCodeCamp/chapter/pull/423#issuecomment-719989846).

These examples of url fields are described in #378 and were added via issues #380 and #381

### PR DESCRIPTION
**Pull request by Ravichandra-C**
- [x] I have read [Chapter's contributing guidelines](https://github.com/freeCodeCamp/chapter/blob/main/CONTRIBUTING.md).
- [x] My pull request has a descriptive title (not a vague title like `Update README.md`).
- [x] My pull request targets the `main` branch of Chapter.

Closes #430 

1. Added type prop to input component in the Event form page
2.removed the DefaultValue prop from input component since it is already
  getting set via UseForm react hook
3. changed the type of URL and Video URL fields to url
4. Added validation decorators to url and video url fields in **server\src\controllers\Events\inputs.ts** file.
5. Enabled class validations in app.ts

### CODE REVIEW

**Review by ojeytonwilliams — CHANGES_REQUESTED** [review]
Thanks a ton for this @Ravichandra-C.  I found a couple of minor things, but it's looking great.

File: server/src/app.ts [hunk 1]
```diff
@@ -26,7 +26,7 @@ export const main = async (app: Express) => {
   app.use(cors({ credentials: true, origin: true }));
   app.use(userMiddleware);
 
-  const schema = await buildSchema({ resolvers, validate: false });
+  const schema = await buildSchema({ resolvers, validate: true });
```
[the remark this answers is not in the document]
Comment (Ravichandra-C) [reply server/src/app.ts]: @ojeytonwilliams Sure, I'll update and commit the changes.

File: cypress/integration/dashboard/events.js [hunk 2]
```diff
@@ -38,14 +38,15 @@ describe('events dashboard', () => {
...
     cy.findByRole('textbox', { name: 'Tags (separated by a comma)' }).type(
       'Test, Event, Tag',
     );
     // TODO: it shouldn't be necessary to clear the date textboxes - the page needs to
     // be fixed
-    cy.findByRole('textbox', { name: 'Start at' }).clear().type(fix.startAt);
-    cy.findByRole('textbox', { name: 'End at' }).clear().type(fix.endAt);
+    cy.get('input[name=start_at]').clear().type(fix.startAt);
+
+    cy.get('input[name=ends_at]').clear().type(fix.endAt);
```
[the remark this answers is not in the document]
Comment (Ravichandra-C) [reply cypress/integration/dashboard/events.js]: @ojeytonwilliams I've made the suggested changes . Could you please review and merge it?

File: server/src/app.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (Ravichandra-C) [reply server/src/app.ts]: @ojeytonwilliams ,I've made the requested changes. Could you please review?

**Review by ojeytonwilliams — APPROVED** [review]
LGTM :+1: thanks @Ravichandra-C!

**Comment by ojeytonwilliams** [comment]
@Zeko369 this looks solid to me, but do you want to take a look?

**Comment by ojeytonwilliams** [comment]
Let's move things along.  Thanks again for putting this together @Ravichandra-C!

**Pull request merged**